src/puppet-dom.js

```diff
@@ -14,7 +14,10 @@
     var onDataReady = options.callback;
     this.element = options.listenTo || document.body;
     var clickHandler = this.clickHandler.bind(this);
-    var historyHandler = this.historyHandler.bind(this);
+    this.historyHandler = this.historyHandler.bind(this);
+
+    /* no need to wait for data to be ready to handle callback */
+    this.element.addEventListener('puppet-redirect-pushstate', this.historyHandler);
```

Comment:
I run than on `fix-redirect-handler` but doesn't really matter, just apply the fix (https://github.com/Palindrom/palindrom-redirect/issues/4#issuecomment-282276197) and run the tests.